src/pages/api/v1/youfocus/index.ts

```diff
@@ -0,0 +1,205 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { connectDB } from '@/middlewares';
+import { apiStatusCodes } from '@/constant';
+import { Playlist } from '@/database';
+import { Video } from '@/interfaces';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  await connectDB();
+  const { method } = req;
+
+  switch (method) {
+    case 'POST':
+      return handleAddPlaylist(req, res);
+    case 'GET':
+      return handleGetAllPlaylists(req, res);
+    default:
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: `Method ${method} not allowed`,
+      });
+  }
+};
+
+// Function to handle adding a new playlist
+const handleAddPlaylist = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    const { playlistUrl } = req.body;
+
+    if (!playlistUrl) {
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: 'Playlist URL is required',
+      });
+    }
+
+    // Extract playlist ID from the URL
+    const playlistId = extractPlaylistId(playlistUrl);
+    if (!playlistId) {
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: 'Invalid playlist URL',
+      });
+    }
+
+    // Check if the playlist ID already exists in the database
+    const existingPlaylist = await Playlist.findOne({ playlistId });
+    if (existingPlaylist) {
+      return res.status(apiStatusCodes.OKAY).json({
+        success: false,
+        message: 'Playlist already exists in the database',
+      });
+    }
+
+    // Fetch playlist data using YouTube Data API
+    const youtubeData = await fetchPlaylistData(playlistUrl);
+
+    if (!youtubeData) {
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: 'Failed to fetch playlist data from YouTube',
+      });
+    }
+
+    // Create and save the new playlist with video details
+    const newPlaylist = await Playlist.create({
+      playlistId,
+      playlistName: youtubeData.playlistName,
+      description: youtubeData.description,
+      videos: youtubeData.videos,
+    });
+
+    return res.status(apiStatusCodes.RESOURCE_CREATED).json({
+      success: true,
+      message: 'Playlist added successfully',
+    });
+  } catch (error) {
+    return res.status(apiStatusCodes.INTERNAL_SERVER_ERROR).json({
+      success: false,
+      message: 'Failed to add playlist',
+    });
+  }
+};
+
+// Function to fetch playlist metadata
+const fetchPlaylistMetadata = async (playlistId: string) => {
+  try {
+    const metadataResponse: any = await fetch(
+      `https://www.googleapis.com/youtube/v3/playlists?part=snippet&id=${playlistId}&key=${process.env.YOUTUBE_API_KEY}`
+    );
+    const metadata = await metadataResponse.json();
+
+    if (!metadataResponse.ok) {
+      throw new Error(`Failed to fetch playlist metadata: ${metadata.error.message}`);
+    }
```

Comment:
All of these functions are for one App YouFocus, We can put them into Utils folder. We've a functions.ts file there we can put.